criu/pie/restorer.c

```diff
@@ -328,10 +328,18 @@ static int restore_creds(struct thread_creds_args *args, int procfd,
 
 static inline int restore_pdeath_sig(struct thread_restore_args *ta)
 {
-	if (ta->pdeath_sig)
-		return sys_prctl(PR_SET_PDEATHSIG, ta->pdeath_sig, 0, 0, 0);
-	else
+	int ret;
+
+	if (!ta->pdeath_sig)
 		return 0;
+
+	ret = sys_prctl(PR_SET_PDEATHSIG, ta->pdeath_sig, 0, 0, 0);
+	if (ret) {
+		pr_err("Unable to set PR_SET_PDEATHSIG(%d): %d\n", ta->pdeath_sig, ret);
+		return -1;
```

Comment:
You could remove this line and just `return ret` below. Same in the other function.

Reply:
It will be correct too, but I like my way a bit more.